File: Jsp/worksapce/6 JavaScript/WebContent/calc.js
/**
 * 객체 만들기
 */
// 빈 객체의 선언		{}: 블록으로 묶어주는 기능
var calc = {};
// 멤버변수 추가
calc.x = 0;
calc.y = 0;
// 멤버함수 추가
calc.setValue = function(p1, p2) {		// 파라미터 값을 멤버변수에 대입
	this.x = p1;
	this.y = p2;
}

calc.plus = function() {
	return this.x + this.y;
}

calc.minus = function() {
	return this.x - this.y;
}

calc.result = function() {
	var value1 = this.plus();
	var value2 = this.minus();
	
	document.write("<p>덧셈결과 : " + value1+ "</p>");
	document.write("<p>뺄셈결과 : " + value2+ "</p>");
}

File: Java/workspace/20-OtherClassType/src/exam6/Calc.java
package exam6;

public class Calc {
	// singleton 패턴 시작
	private static Calc instance = null;

	public static Calc getInstance() {
		if (instance == null)
			instance = new Calc();
		return instance;
	}

	// 기본 생성자를 private으로 하면, new를 통한 객체 생성이 금지된다.
	private Calc() {
	}

	// singleton 패턴 끝
	public void plus(int x, int y) {
		System.out.println(x + " + " + y + " = " + (x + y));

	}

	public void minus(int x, int y) {
		System.out.println(x + " - " + y + " = " + (x - y));

	}

}
